### PR TITLE
Replace absolute links to specs / proposals to relative links

### PR DESCRIPTION
**Pull request by BillWagner**
Now that we publish the C# specs and proposals on docs, these links should point to the content published on docs rather than the GitHub source.

Relies on #11190 

There will be 1 warning until that PR is merged.

### CODE REVIEW

**Review by rpetrusha — APPROVED** [review]
LGTM, @BillWagner. You can merge this after you merge #11190.

**Comment by BillWagner** [comment]
closing and reopening to verify the build.

**Pull request merged**
